src/api/mutations.ts

```diff
@@ -322,3 +330,113 @@ export const groupMutations = {
       },
     }),
 };
+
+// 게시글 좋아요 뮤테이션
+export const likeMutations = {
+  // 좋아요 추가
+  likeMutation: (articleId: number) => ['article', articleId], // 게시글 상세 쿼리 키에 좋아요 데이터가 있으므로 해당 키 사용
+  likeMutationOptions: ({
+    articleId,
+    queryClient,
+  }: {
+    articleId: number;
+    queryClient: QueryClient;
+  }) =>
+    mutationOptions({
+      mutationFn: (articleId: number) => likeArticle(articleId),
+      onMutate: async () => {
+        await queryClient.cancelQueries({
+          queryKey: boardQueries.article(articleId),
+        });
+        const prevData = queryClient.getQueryData(
+          boardQueries.article(articleId),
+        );
+
+        // 캐시 업데이트
+        queryClient.setQueryData<ArticleDetailResponse>(
+          boardQueries.article(articleId),
+          old => {
+            if (!old) return old;
+
+            const newLikeCount = old.isLiked
+              ? old.likeCount - 1
+              : old.likeCount + 1;
+
+            return {
+              ...old,
+              likeCount: newLikeCount,
+              isActive: !old.isLiked,
+            };
+          },
+        );
+
+        return { prevData }; // 실패 시 복구되도록 이전 데이터 반환
+      },
+
+      onError: (_err, _variables, context) => {
+        if (context?.prevData) {
+          queryClient.setQueryData(
+            boardQueries.article(articleId),
+            context.prevData,
+          );
+        }
+      },
+
+      onSettled: () => {
+        queryClient.invalidateQueries({
+          queryKey: boardQueries.article(articleId),
+        });
+      },
+    }),
```

Comment:
여기와 unlike 쪽에도 태경님이 공유해 주셨던 조건 적용하면 좋을 것 같습니다! 
```suggestion
      onSettled: () => {
        if (queryClient.isMutating({ mutationKey: likeMutations.likeMutation(articleId) }) === 1) {
          queryClient.invalidateQueries({
          queryKey: boardQueries.article(articleId),
          });
        }
      },
    }),
```

Reply:
저도 태경님이랑 같은 생각이었는데 일단 조건을 추가해두긴 했습니다.(제안하신 이유가 있을 것 같아서ㅎ) 근데 궁금한 게 있는데 혹시 unlikeMutation에만 isMutating 조건을 추가해야되는 이유가 있나요? 해당 조건을 추가하려면 likeMutation에도 같이 추가해야 될 것 같단 생각이 들고, unlikeMutation에서 조건 없이 무효화되도록 했을 때와 동작에 별 차이가 없는 것 같아요..! @khuyjh

Reply:
사실 일반적인 사용자 동작을 생각했을 때 좋아요 버튼 하나 있는걸 연속으로 수십번 클릭할 일은 거의 없겠지만.. 애초에 해당 조건문 처리가 아주 예외적인 케이스 방어를 위한 코드로 알고 있고, 하나의 대상에 연속적인 뮤테이션을 실행할 때에도 발생가능성이 없지는 않은 것 같아서 저는 있어도 나쁠 것은 없는 코드정도로 생각하고 제안 드리긴 했습니다. 사용자 경험면에서 없더라도 불편함을 크게 느낄 것 같지는 않아요!
그리고 수정 제안한 코드쪽이 likeMutation쪽이고 이 코드를 unlikeMutation 쪽에도 추가하는게 좋겠다고 적은 것이라 만약 적용한다면 양쪽 다 적용하는게 좋을 것 같다는 의미였습니다! 이 부분은 전달 미스가 있었던 것 같네요😅